src/Particle/ParticleSet.cpp

```diff
@@ -390,6 +390,33 @@ void ParticleSet::update(bool skipSK)
   activePtcl = -1;
 }
 
+void ParticleSet::flex_update(const RefVector<ParticleSet>& p_list, bool skipSK)
+{
+  if (p_list.size() > 1)
+  {
+    ScopedTimer update_scope(p_list[0].get().myTimers[PS_update]);
+
+    for (ParticleSet& pset : p_list)
+      pset.setCoordinates(pset.R);
+
+    auto& dts = p_list[0].get().DistTables;
+    for (int i = 0; i < dts.size(); ++i)
+    {
+      const auto dt_list(extractDTRefList(p_list, i));
+      dts[i]->mw_evaluate(dt_list, p_list);
+    }
+
+    if (!skipSK && p_list[0].get().SK)
+    {
+#pragma omp parallel for
```

Comment:
This is a consistent design in all the batched interfaces.
Inside a flex_XXX interface, it either calls mw_XXX to reach the next layer of abstraction.
See TrialWaveFunction and WaveFunctionComponent relation.
In the fallback mw_XXX in WaveFunctionComponent, a loop over walkers and `omp parallel for` are added to handle next level threading over walkers.
If adding mw_XXX needs too many lines of code and only CPU version is needed, a direct loop over walkers with `omp parallel for` is put in place.
The reason adding this "parallel for" is to run either 1 crowd with many walkers over OpenMP threads or many crowds with many walkers using OpenMP nested threads.

How to abstract the next OpenMP level with nice C++ is separate topic and I prefer to look at it when we have more spare effort.

Here the SK(StructFact) needs a batched evaluation but the class is not yet refactored. For this reason you see the bare loop and omp pragmas. This is fine since only CPU version is needed at the moment. If we need to deal with offload, the mw_XXX interfaces to StructFact must be made available.